gnd/src/main/java/com/google/android/gnd/ui/map/Extent.java

```diff
@@ -0,0 +1,69 @@
+package com.google.android.gnd.ui.map;
+
+import androidx.annotation.Nullable;
+
+import com.google.android.gnd.model.basemap.tile.Tile;
+import com.google.auto.value.AutoValue;
+
+@AutoValue
+public abstract class Extent {
+  public enum State {
+    DOWNLOADED,
+    PENDING_DOWNLOAD,
+    PENDING_REMOVAL,
+    NONE
+  }
+
+  public abstract String getId();
+
+  public abstract State getState();
+
+  public static Builder newBuilder() {
+    return new AutoValue_Extent.Builder();
+  }
+
+  public abstract Builder toBuilder();
+
+  public static Extent fromTile(Tile tile) {
+    switch (tile.getState()) {
+      case IN_PROGRESS:
+        return Extent.newBuilder().setId(tile.getId()).setState(State.PENDING_DOWNLOAD).build();
+      case DOWNLOADED:
+        return Extent.newBuilder().setId(tile.getId()).setState(State.DOWNLOADED).build();
+      case FAILED:
+        return Extent.newBuilder().setId(tile.getId()).setState(State.NONE).build();
+      case PENDING:
+        return Extent.newBuilder().setId(tile.getId()).setState(State.PENDING_DOWNLOAD).build();
+      default:
+        return Extent.newBuilder().setId(tile.getId()).setState(State.NONE).build();
+    }
+  }
+
+  @AutoValue.Builder
+  public abstract static class Builder {
+    public abstract Builder setId(String id);
+
+    public abstract Builder setState(State state);
+
+    public abstract Extent build();
+  }
+
+  @Override
+  public boolean equals(@Nullable Object obj) {
```

Comment:
Indeed! Originally I was maintaining sets of Extents rather than extent IDs--the equals override still may not have been necessary there, but I wanted Extents with the same ID to be equivalent (that way we could accurately update state) but I don't know enough about Java's object comparisons, so I implemented this just in case. 

I'm going to wait until everything else is in so that it's more easily verifiable whether we can drop this override. For now I added a TODO to remove it.